src/library_html5_webgl.js

```diff
@@ -586,10 +586,16 @@ function handleWebGLProxying(funcs) {
       delete funcs[i + '__proxy'];
     }
   }
+#else
+  // In single threaded mode just delete our custom __proxy addributes, otherwise
+  // they will causes errors in the JS compiler.
+  for (var i in funcs) {
+    delete funcs[i + '__proxy'];
```

Comment:
This code only runs if `!SHARED_MEMORY` so when building with wasm workers or pthreads this code doesn't run and the attributes are not ignored.

I agree this could probably use a comment.  The issue is that `library_htm5_webgl.js` uses custom `__proxy` addribute `sync_on_webgl_context_handle_thread` which needs to get either re-written (above), or removed (here) before the library is processed, otherwise we get an `Invalid proxyingMode` jsifier.

Reply:
> This code only runs if !SHARED_MEMORY

I am confused. Taking a closer look, the code runs if

```js
if (SHARED_MEMORY) {
  if (PTHREADS) {
    // ...
  } else if (WASM_WORKERS && ASSERTIONS) {
    // emit "assert(!ENVIRONMENT_IS_WASM_WORKER"
  }
}
```
but I think I wrote that when I still had the assumption that PTHREADS and WASM_WORKERS would always be mutually exclusive and you'd only build targeting one or the other - but turns out targeting both at the same time made sense.

So that above code should instead be

```js
if (SHARED_MEMORY) {
  if (PTHREADS) {
    // ...
  }
  if (WASM_WORKERS && ASSERTIONS) {
    // emit "assert(!ENVIRONMENT_IS_WASM_WORKER"
  }
}
```
so that in PTHREADS + WASM_WORKERS builds the assertion is also added. Would that appropriately resolve the issue?

I still think no library_x.js file should need to mutate the __proxy fields depending on the build mode.

Reply:
I think I see what you are saying.  We should probably not erase the __proxy attributes in the wasm workers build.  I will create a followup to do that.

I agree the custom proxy stuff in `library_html5_webgl.js`  is unusual but it looks like you were the one that created this mechanism back in #11356, and I think its OK as long as it get locally handled/erased.

Reply:
Sorry, I mis-remembered, I already did do that in this PR, that `__proxy` handling is enabled here for all `SHARED_MEMORY` builds.

I think updating jsifier.js in way you suggest sounds fine to me.  That seems like a pre-existing issue that is somewhat unrelated to this PR and would effect all libraries, right?

Are you happy with this PR in the way it deals with `library_html5_webgl.js`?

Reply:
> That seems like a pre-existing issue that is somewhat unrelated to this PR and would effect all libraries, right?

Yeah, I think it would be preferable to have that affect all libraries, so that any developer who is calling a `__proxy: sync` function inside a Wasm Worker would get an assertion firing on them. That is because I don't think any function with that label would work properly when called in the context of a Wasm Worker.